src/hooks/useAsyncFetch.js

```diff
@@ -0,0 +1,18 @@
+import { useProducts } from "../context/product-context";
+import {useEffect} from "react";
+import axios from "axios";
+
+export function useAsyncFetch({url, dispatchType, dispatchPayload}){
+    const {dispatch}  = useProducts();
+    useEffect(()=>{
+        (async function asyncProductFetch(){
+            try{
+                const {data} = await axios.get(url);
+                console.log("Calling api"+url);
+                dispatch({type:dispatchType, payload: data[dispatchPayload]});
+            }catch(err){
+                console.log(err);
+            }
+        })();
+    },[]);
+}
```

Comment:
Loved that you have created a custom hook for all API calls.

Reply:
Thank you